ui/plugins/pinboard.js

```diff
@@ -84,6 +84,12 @@ treeherder.controller('PinboardCtrl', [
 
         $scope.save = function() {
             var errorFree = true;
+            if ($scope.enteringBugNumber) {
+                // we should save this for the user, as they likely
+                // just forgot to hit enter. Returns false if invalid
+                thNotify.send("Please enter a valid bug number", "danger");
+                errorFree = $scope.saveEnteredBugNumber();
+            }
```

Comment:
Changed this so it forces the bug number to be saved prior to checking anything else if we're in enteringBugNumber mode. This lets us validate the bug number more easily. If it isn't a valid positive integer, it warns the user about it and bails on trying to submit the classification.